### DISCUSSION TITLE
Hacked accounts

### DISCUSSION THREAD

**Discussion by ghost** [deso-protocol/dips#62] [Medium CIPs]
Currently, Bitclout has very few users. But after the marketing efforts, millions of users will join us in a short time. And that will bring some problems. One of them is the possibility of hacking accounts. In this case, users should be able to protect their own account. my suggestions;

- Account migration feature just like reserve accounts = change seed words
- One more layer of security can be added

**Comment by tijno** [comment]
Could you maybe elaborate on what you think the best approach is for each these suggestions ?

**Comment by ItsAditya-xyz** [comment]
i don't think changing seed words is even possible. Yeah maybe we can have some extra layer of security like passphrase (as we have in Cloutfeed App) or 2FA

**Reply by ghost** [reply]
The method applied in reserve accounts can be applied.  In this way, seed words will be changed indirectly.

**Reply by sull** [reply]
are you referring to SwapIdentity?

I believe this superadmin utility may be modified for use by noderunners.

**Comment by iPaulPro** [comment]
"Along with this work is implementing a way for ordinary users to transfer their profile to a new key, which will allow us to hopefully remove/sunset SwapIdentity."

According to the FAQ we'll soon have the ability to do swaps ourselves.

https://docs.bitclout.com/faq/bitclout-faq#can-we-have-more-visibility-of-the-product-roadmap-this-will-help-the-dev-community-to-build-appropriately-would-you-consider-making-the-roadmap-public-genegmb-davidsun-usmansheikh-cloutcurator

**Comment by sull** [comment]
> Currently, Bitclout has very few users. But after the marketing efforts, millions of users will join us in a short time. And that will bring some problems. One of them is the possibility of hacking accounts. In this case, users should be able to protect their own account. my suggestions;
> 
> * Account migration feature just like reserve accounts = change seed words
> * One more layer of security can be added

* SwapIdentity may be leveraged after it is modified to not be a superadmin tool that currently is only available to the bitclout core dev team, as I understand it. This will allow node admins to transfer a username to a new public address if there are cases where a seed phrase is lost or compromised.

* Besides using Google SignIn, MFA/2FA can be used in front of the seed phrase login and/or for certain transaction types.
